src/Sentry/Debouncer.cs

```diff
@@ -0,0 +1,108 @@
+
+    private Debouncer(DebouncerInterval intervalType, int eventMaximum = 1, TimeSpan? cooldown = null)
+    {
+        _intervalType = intervalType;
+        _cooldown = cooldown;
+        _eventMaximum = eventMaximum;
+    }
+
+    /// <summary>
+    /// Creates a debouncer that limits the number of events per minute
+    /// </summary>
+    /// <param name="eventMaximum">The maximum number of events that will be processed per minute</param>
+    /// <param name="cooldown">An optional obligatory cooldown since the last event before any other events will be processed</param>
+    /// <returns></returns>
+    public static Debouncer PerMinute(int eventMaximum = 1, TimeSpan? cooldown = null)
+        => new(DebouncerInterval.Minute, eventMaximum, cooldown);
+
+    /// <summary>
+    /// Creates a debouncer that limits the number of events per hour
+    /// </summary>
+    /// <param name="eventMaximum">The maximum number of events that will be processed per hour</param>
+    /// <param name="cooldown">An optional obligatory cooldown since the last event before any other events will be processed</param>
+    /// <returns></returns>
+    public static Debouncer PerHour(int eventMaximum = 1, TimeSpan? cooldown = null)
+        => new(DebouncerInterval.Hour, eventMaximum, cooldown);
+
+    /// <summary>
+    /// Creates a debouncer that limits the number of events per day
+    /// </summary>
+    /// <param name="eventMaximum">The maximum number of events that will be processed per day</param>
+    /// <param name="cooldown">An optional obligatory cooldown since the last event before any other events will be processed</param>
+    /// <returns></returns>
+    public static Debouncer PerDay(int eventMaximum = 1, TimeSpan? cooldown = null)
+        => new(DebouncerInterval.Day, eventMaximum, cooldown);
+
+    /// <summary>
+    /// Creates a debouncer that limits the number of events that will be processed for the lifetime of the application
+    /// </summary>
+    /// <param name="eventMaximum">The maximum number of events that will be processed</param>
+    /// <param name="cooldown">An optional obligatory cooldown since the last event before any other events will be processed</param>
+    /// <returns></returns>
+    public static Debouncer PerApplicationLifetime(int eventMaximum = 1, TimeSpan? cooldown = null)
+        => new(DebouncerInterval.ApplicationLifetime, eventMaximum, cooldown);
+
+    private TimeSpan IntervalTimeSpan()
+    {
+        switch (_intervalType)
+        {
+            case DebouncerInterval.Minute:
+                return TimeSpan.FromMinutes(1);
+            case DebouncerInterval.Hour:
+                return TimeSpan.FromHours(1);
+            case DebouncerInterval.Day:
+                return TimeSpan.FromDays(1);
+            case DebouncerInterval.ApplicationLifetime:
+                return TimeSpan.MaxValue;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(_intervalType));
```

Comment:
I would like to say this would only be on Init... but the class and the static factory methods in it are public - so people could try to use it in unusual ways (e.g. to debounce events in their apps that have nothing to do with Heap Dumps). In those instances though, we're no longer dealing with the Sentry SDK proper, but their application code... and we do want to throw exceptions for problems if this class is used in their application code right?